feat 采购模块

fishSteak opened the pull request:
1、采购申请(完成)
2、采购订单，审核、开关、采购（完成）。入库联动(完成)，出库联动+付款联动。（完成）
3、入库（完成）+出库+付款。(完成)
4、供应商（完成）+财务主体（完成）

goodman59 committed:
feature 2025/01/06  采购模块

fishSteak committed:
fix 采购单修改属性

feature 采购订单entity

fishSteak committed:
fix 采购单修改属性

fishSteak committed:
fix 采购单修改属性

fishSteak committed:
fix 临时保存一下，修改后的VO，entity

fishSteak committed:
fix Erp的CRUD初步完成，状态待修改。

fishSteak committed:
fix buildPurchaseOrderVOPageResult 修改状态

fix buildPurchaseOrderVOPageResult 修改

fishSteak committed:
fix 入库单VO的修改完成

fishSteak committed:
fix 入库单修改完成

fishSteak committed:
fix 提取公共类ErpPurchaseBaseRespVO

fishSteak committed:
fix 修改类型

fishSteak committed:
fix 修改属性1

fix 修改属性

fishSteak committed:
fix 修改属性3

fishSteak committed:
fix 修改采购属性。

fishSteak committed:
fix 去掉空格

Commit:
fix 修改CRUD

Commit:
fix 新增、修改、删除

Commit:
fix 新增、修改、删除

Commit:
fix 设置状态、小修改

Commit:
fix 设置状态、小修改

fishSteak committed:
fix 采购申请修改

fishSteak committed:
fix 增加了状态字段给mysql

fishSteak committed:
Merge branch 'somle-main-purchase' into somle-main-pirchase-request

# Conflicts:
#	yudao-module-erp/yudao-module-erp-biz/pom.xml
#	yudao-module-erp/yudao-module-erp-biz/src/main/java/cn/iocoder/yudao/module/erp/controller/admin/purchase/ErpPurchaseRequestController.java
#	yudao-server/src/main/resources/application-test.yaml

fishSteak committed:
fix 增加了状态字段给mysql

fishSteak committed:
fix 测试注释

fishSteak committed:
fix 状态2字段修改

fishSteak committed:
fix 采购申请单初步完成。合并订单待完成。

fishSteak committed:
Merge branch 'refs/heads/somle-main' into somle-main-pirchase-request

# Conflicts:
#	yudao-server/src/main/resources/application-test.yaml

fishSteak committed:
Merge branch 'somle-main' into somle-main-pirchase-request

# Conflicts:
#	yudao-module-erp/yudao-module-erp-biz/src/test/java/cn/iocoder/yudao/module/erp/service/logistic/customrule/ErpCustomRuleServiceImplTest.java

fishSteak committed:
Merge branch 'somle-main' into somle-main-purchase-request

fishSteak committed:
Merge branch 'somle-main' into somle-main-purchase-request

# Conflicts:
#	yudao-module-erp/yudao-module-erp-biz/pom.xml

fishSteak committed:
Merge branch 'somle-main' into somle-main-purchase-request

fishSteak committed:
Merge branch 'somle-main' into somle-main-purchase-request

# Conflicts:
#	yudao-module-erp/yudao-module-erp-api/src/main/java/cn/iocoder/yudao/module/erp/enums/ErpAuditStatus.java
#	yudao-server/src/main/resources/application-test.yaml

fishSteak committed:
fix 修改

fishSteak committed:
fix 修改

fishSteak committed:
fix 修改一下逻辑

fishSteak committed:
Merge branch 'somle-main' into somle-main-purchase-request

# Conflicts:
#	yudao-module-erp/yudao-module-erp-biz/src/main/java/cn/iocoder/yudao/module/erp/controller/admin/purchase/ErpPurchaseRequestController.java
#	yudao-module-erp/yudao-module-erp-biz/src/main/java/cn/iocoder/yudao/module/erp/controller/admin/purchase/vo/request/ErpPurchaseRequestRespVO.java
#	yudao-module-erp/yudao-module-erp-biz/src/main/java/cn/iocoder/yudao/module/erp/dal/mysql/logistic/category/item/ErpCustomCategoryItemMapper.java

fishSteak committed:
fix 状态机测试类通过

fix 添加状态机

fishSteak committed:
fix 修改

Commit:
fix 优化

fishSteak committed:
fix 修改

fishSteak committed:
fix 修改财务模块-主体管理

fix 增加财务模块-主体管理

Commit:
fix 优化校验逻辑

fishSteak committed:
fix 采购按钮待完成

fishSteak committed:
fix 采购异常

fishSteak committed:
fix 添加审核申请接口

fishSteak committed:
fix 优化

fishSteak committed:
fix 数据库异常封装

fishSteak committed:
fix 修改

fishSteak committed:
fix 等待增加item的订购状态的状态机。

fishSteak committed:
fix 增加了异常回调函数，中文

fishSteak committed:
fix 优化

fishSteak committed:
fix 采购申请告一段落。开始采购采购单状态

fishSteak committed:
fix 采购订单修改

Commit:
fix 优化

fishSteak committed:
feature 新增了供应商管理字段,验证通过

fishSteak committed:
fix 修改

fishSteak committed:
fix 完成财务主体缓存调用

fishSteak committed:
fix 缓存优化、最终样式

fishSteak committed:
fix 缓存优化、最终样式

fishSteak committed:
fix 采购申请合并测试，逻辑完成

fishSteak committed:
fix 采购申请合并

Commit:
fix 优化-采购状态+数量，的逻辑

Commit:
Merge branch 'somle-main' into somle-main-purchase-request

Commit:
fix 修改订单子项目状态机 ; 添加入库单状态联动备注项,order审核状态主表,状态机完善了

Commit:
feat: 完善ERP执行状态机和支付状态机，添加状态转换事件和注释

Commit:
feat(erp): 完善采购订单执行状态机

1. 新增 ErpExecutionStatus 执行状态枚举
- 定义待执行、执行中、已完成等7种状态
- 实现状态转换和验证逻辑

2. 完善支付状态机实现
- 修复 ActionOrderPayImpl 依赖注入问题
- 优化 ErpPaymentStatus 状态定义

3. 增强状态机事件定义
- 在 ErpEventEnum 中新增执行相关事件
- 完善事件注释和业务说明

Commit:
fix 修改付款状态传递事件

fishSteak committed:
迁移状态机基本异常类回调方法。

fishSteak committed:
fix 采购申请合并

fishSteak committed:
fix 修改

fishSteak committed:
fix 修改

fishSteak committed:
fix 修改

fishSteak committed:
fix 修改

fishSteak committed:
fix 订单完成，入库数据，付款数据待完成

fishSteak committed:
fix 修改未订购返回数量

fishSteak committed:
fix 入库单修改

c-tao committed:
feature: 提取yudao-module-tms模块，但是需要修复对erp产品的引用

fishSteak committed:
fix 入库单修改

fishSteak committed:
fix 采购订单修改，保证入库单的所有字段确实符合要求

fishSteak committed:
fix 采购订单修改, merge修改,舍弃缓存,入库单字段核对通过

fishSteak committed:
fix 根据DTO来确定表名，通过表名来生成对应的sql。

fishSteak committed:
fix 备注

fishSteak committed:
Revert "fix 备注"

This reverts commit 8daf081605712e9f4bc321e25174eb2e5f558006.

fishSteak committed:
Revert "fix 根据DTO来确定表名，通过表名来生成对应的sql。"

This reverts commit f572a7fcced4c429f5e3472000af51a9a923bf71.

fishSteak committed:
fix 修改策略。
1. 规则-{限定产品ids筛选通过api拿} 得到DO
2. 规则{限定产品ids筛选通过api拿}->分类子表(国别=规则国别)->分类。  得到BO

fishSteak committed:
fix 完成修复，待全局测试！！

fishSteak committed:
fix 格式整理

fishSteak committed:
fix erp->tms

fishSteak committed:
fix 修改部分接口魔法值

fishSteak committed:
fix 修改

fishSteak committed:
fix 测试完成

fishSteak committed:
fix gumaomao写的缓存 我改了一下

fishSteak committed:
fix 入库修改了

fishSteak committed:
fix Dockerfile 修改了 更加人性化。

fishSteak committed:
fix 实现了入库单联动订单项->联动采购申请项

fishSteak committed:
fix 1.新增申请单+申请项属性,in_status 入库状态。
2.入库->采购->申请，完成采购申请项的入库修改。

fishSteak committed:
fix bug修改完成

fishSteak committed:
fix 入库单的增改差检查完毕

fishSteak committed:
fix 入库单检查完成，前端待实现

fishSteak committed:
fix 采购申请项，去除申请项的采购订单关联。由订单项负责维护申请项的关系。申请项：订单项 == 一对多。

fishSteak committed:
fix 采购申请项，去除申请项的采购订单关联。由订单项负责维护申请项的关系。申请项：订单项 == 一对多。

fishSteak committed:
Merge remote-tracking branch 'origin/somle-main-purchase-request' into somle-main-purchase-request

fishSteak committed:
fix 修改订单合并入库单接口

fishSteak committed:
Merge branch 'somle-main-taoc' into somle-main-purchase-request

# Conflicts:
#	yudao-module-erp/yudao-module-erp-api/src/main/java/cn/iocoder/yudao/module/erp/enums/ErrorCodeConstants.java
#	yudao-module-erp/yudao-module-erp-biz/pom.xml
#	yudao-module-erp/yudao-module-erp-biz/src/main/java/cn/iocoder/yudao/module/erp/service/product/ErpProductServiceImpl.java

fishSteak committed:
fix tms的产品分类服务，缓存修改。erp产品的缓存修改。

fishSteak committed:
Merge branch 'somle-test' into somle-main-purchase-request

# Conflicts:
#	pom.xml
#	somle-eccang/src/main/java/com/somle/eccang/service/EccangService.java
#	yudao-framework/yudao-common/src/main/java/cn/iocoder/yudao/framework/common/util/web/WebUtils.java
#	yudao-module-erp/yudao-module-erp-biz/src/test/java/cn/iocoder/yudao/module/erp/dal/mysql/logistic/category/ErpCustomCategoryMapperTest.java
#	yudao-module-erp/yudao-module-erp-biz/src/test/java/cn/iocoder/yudao/module/erp/dal/mysql/logistic/category/item/ErpCustomCategoryItemMapperTest.java
#	yudao-module-erp/yudao-module-erp-biz/src/test/java/cn/iocoder/yudao/module/erp/dal/mysql/logistic/customrule/ErpCustomRuleMapperTest.java
#	yudao-module-erp/yudao-module-erp-biz/src/test/java/cn/iocoder/yudao/module/erp/service/ErpSyncDataTest.java
#	yudao-module-erp/yudao-module-erp-biz/src/test/java/cn/iocoder/yudao/module/erp/service/logistic/category/ErpCustomCategoryServiceImplTest.java
#	yudao-module-erp/yudao-module-erp-biz/src/test/java/cn/iocoder/yudao/module/erp/service/logistic/category/item/ErpCustomCategoryItemServiceImplTest.java
#	yudao-module-erp/yudao-module-erp-biz/src/test/java/cn/iocoder/yudao/module/erp/service/logistic/customrule/ErpCustomRuleServiceImplTest.java
#	yudao-module-erp/yudao-module-erp-biz/src/test/java/cn/iocoder/yudao/module/erp/service/purchase/ErpSupplierProductServiceImplTest.java
#	yudao-server/src/main/resources/application-test.yaml

fishSteak committed:
fix ErpSupplierServiceImpl集合查询null校验

fishSteak committed:
fix ErpSupplierServiceImpl集合查询null校验,修复了入库单-订单的数量变更bug，使用add方法。

fishSteak committed:
fix 付款状态机完成，主付款状态机传递事件待完成。退货单关联待完成。

fishSteak committed:
fix 完结入库单的修改，出库+合并出库

fishSteak committed:
fix 修复

fishSteak committed:
fix // 第二步，批量添加、修改、删除 需要联动退货数量

fishSteak committed:
Merge branch 'somle-main' into somle-main-purchase-request

# Conflicts:
#	yudao-module-erp/yudao-module-erp-api/src/main/java/cn/iocoder/yudao/module/erp/enums/ErrorCodeConstants.java
#	yudao-module-erp/yudao-module-erp-biz/pom.xml
#	yudao-module-erp/yudao-module-erp-biz/src/main/java/cn/iocoder/yudao/module/erp/dal/redis/no/ErpNoRedisDAO.java
#	yudao-module-erp/yudao-module-erp-biz/src/main/java/cn/iocoder/yudao/module/erp/service/product/ErpProductServiceImpl.java
#	yudao-module-tms/yudao-module-tms-biz/src/main/java/cn/iocoder/yudao/module/tms/dal/mysql/logistic/customrule/ErpCustomRuleMapper.java

fishSteak committed:
fix 退货单待校验

fishSteak committed:
fix 修改

fishSteak committed:
fix 修改

fishSteak commented:
![image](https://github.com/user-attachments/assets/86602487-e695-4f0b-bbcd-3f4f50d2a973)
模块ER图

fishSteak commented:
![image](https://github.com/user-attachments/assets/75d8947d-71cf-42f2-a1f3-e0c49b74880f)
数据联动图

fishSteak committed:
fix 修改

fishSteak committed:
fix 采购推虎检查完毕。

fishSteak committed:
fix 禁止定时任务

fishSteak committed:
fix 禁止定时任务

fishSteak committed:
fix 修复

fishSteak committed:
Resolved merge conflict

fishSteak committed:
fix temp

fishSteak committed:
fix 合并修改

fishSteak committed:
fix 修改了对于 channel 的配置。

fishSteak committed:
fix 修改

fishSteak committed:
fix 优化

c-tao commented on yudao-module-tms/yudao-module-tms-api/src/main/java/cn/iocoder/yudao/module/tms/api/logistic/customrule/dto/ErpCustomRuleDTO.java:
```diff
@@ -1,5 +1,5 @@
 package cn.iocoder.yudao.module.tms.api.logistic.customrule.dto;
-import cn.iocoder.yudao.module.erp.api.product.dto.ErpProductDTO;
+import cn.iocoder.yudao.module.srm.api.product.dto.ErpProductDTO;
```
包名改错了

fishSteak replied:
是tmsProductDTO吗

fishSteak committed:
fix 改了包名

fishSteak committed:
fix 改了包名

fishSteak committed:
撤销了 "fix 修改了对于 channel 的配置。"

This reverts commit 2b0d77cd

fishSteak committed:
@Autowired注解替换@Resource，对于channel的调用。优先使用名称去注入
fix 修复了错误

fishSteak committed:
fix 修改

fishSteak committed:
fix word生成完成

fishSteak committed:
fix pdf转换成功

fishSteak committed:
fix 从本地resource生成word

fishSteak committed:
fix 接口排错

fishSteak committed:
fix 修复

fishSteak committed:
fix 修改申请单+采购订单

fishSteak committed:
fix 修改

fishSteak committed:
fix 调整付款状态的联动，入库项->订单项

fishSteak committed:
fix 给生成合同模板添加 缓存预热，放入AOP中

fishSteak committed:
fix 修改

fishSteak committed:
fix 修改

fishSteak committed:
fix 修复CommonResult 返回pdf异常

fishSteak committed:
fix  拆分fms

c-tao commented on yudao-framework/yudao-spring-boot-starter-test/src/main/java/cn/iocoder/yudao/framework/test/core/ut/BaseDbUnitTest.java:
```diff
@@ -12,34 +12,33 @@
 @SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.NONE, classes = BaseDbUnitTest.Application.class)
 @ActiveProfiles("unit-test") // 设置使用 application-unit-test 配置文件
-@Sql(scripts = "/sql/clean.sql", executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD) // 每个单元测试结束后，清理 DB
+//@Sql(scripts = "/sql/clean.sql", executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD) // 每个单元测试结束后，清理 DB
```
这个不能注，会导致测试环境失败

fishSteak replied:
已撤销

fishSteak committed:
fix  拆分fms

fishSteak committed:
fix 修复idea扫描爆红

fishSteak committed:
fix 修复爆红idea检测异常

fishSteak committed:
fix 修复idea扫描爆红

fishSteak committed:
Merge remote-tracking branch 'origin/somle-main-purchase-request' into somle-main-purchase-request

fishSteak committed:
fix 修复

fishSteak committed:
fix 1、修改合并VO和订单创建的属性。

fishSteak committed:
fix 修改

c-tao committed:
fix: 修复fms模块的类名前缀

c-tao committed:
fix: 修复tms模块的类名前缀

c-tao committed:
fix: 修复srm模块的类名前缀

c-tao committed:
fix: 修复srm模块的类名前缀 2

c-tao committed:
fix: erp模块修剪

c-tao commented on yudao-framework/yudao-spring-boot-starter-mybatis/src/main/java/cn/iocoder/yudao/framework/mybatis/core/vo/BaseVO.java:
```diff
@@ -0,0 +1,47 @@
+package cn.iocoder.yudao.framework.mybatis.core.vo;
```
这个类做什么用

fishSteak replied:
基本返回VO，有创建人 更新人 字段，不用重复写。直接vo继承这个就可以，按需

c-tao replied:
和BaseVO放一个包

fishSteak replied:
这个类就是baseVO

c-tao commented on yudao-module-system/yudao-module-system-api/src/main/java/cn/iocoder/yudao/module/system/api/utils/Validation.java:
```diff
@@ -6,4 +6,13 @@ interface OnCreate {
 
     interface OnUpdate {
     }
+
```
这些validation是所有yudao-module通用的还是只有erp模块会用？

fishSteak replied:
通用

c-tao replied:
工具类放yudao-framework

fishSteak committed:
fix 采购申请单+采购订单已校验。修改no待完成

fishSteak committed:
fix 采购申请修改no已完成

fishSteak committed:
fix 采购申请修改no已完成

fishSteak committed:
fix 采购订单生成NO，修改完成

fishSteak committed:
fix 采购订单完毕

fishSteak committed:
fix 修改

fishSteak committed:
fix 修改

fishSteak committed:
fix 修改

fishSteak committed:
fix 英文word 模板修改

fishSteak committed:
fix 英文word 模板修改

fishSteak committed:
fix 修改

c-tao commented on yudao-server/src/main/resources/application-dev.yaml:
```diff
@@ -52,6 +52,10 @@ spring:
       host: 127.0.0.1 # 地址
       port: 6379 # 端口
       database: 1 # 数据库索引
+  task:
```
这是做什么

fishSteak replied:
@Scheduled  禁止执行，spring:
  task:
    scheduling:
      enabled: false  暂时没有生效

c-tao commented on yudao-server/Dockerfile:
```diff
@@ -3,21 +3,25 @@
 FROM eclipse-temurin:21-jre
```
这些为什么要改

fishSteak replied:
这个dockerfile可视化更好、目的都一样的，yudao写的太罗嗦了

fishSteak replied:
他写死了yudao.jar  到时候改项目名称 打包名称不一样 dockerfile 还得手动改。

fishSteak committed:
fix 表名的修改 erp -> srm

fishSteak committed:
Merge branch 'somle-test' into somle-main-purchase-request

# Conflicts:
#	pom.xml

fishSteak committed:
refactor(srm): 重构采购模块代码

- 优化了 SrmPurchaseInController 中的导入导出逻辑
- 重构了 SrmPurchaseInSaveReqVO，添加了仓库 ID 字段
-调整了 SrmPurchaseInServiceImpl 中的数据插入流程
-统一了采购订单和请求中的数量字段使用 qty
- 在 local 环境中禁用了 Somle 的自动配置
- 更新了 YudaoServerApplication 中的条件注解

fishSteak committed:
refactor(srm): 重构采购订单相关代码

- 修改包名：将 order 相关的 VO 类移动到 order/req 目录下
- 优化导入语句：更新相关类的包名引用
- 调整代码格式：改善部分代码的可读性，如换行、缩进等
- 新增接口：添加更新采购订单 JSON 属性的接口

fishSteak committed:
refactor(srm): 更新采购订单 JSON 属性功能

- 修改了 SrmPurchaseOrderController 中的 updatePurchaseOrderJson 方法调用
- 更新了 SrmPurchaseOrderService接口中方法名
-调整了 SrmPurchaseOrderServiceImpl 中的实现方法
- 优化了代码注释和格式

fishSteak committed:
refactor(srm): 重构采购订单相关代码

- 注释掉 SrmPurchaseOrderDO 中的 xCode 字段
- 移除 SrmPurchaseOrderItemMapper 和 SrmPurchaseOrderMapper 中与 xCode 相关的查询条件
- 更新 SrmPurchaseOrderMapper 中的 getBOWrapper 方法，加入申请人ID和申请部门ID的查询条件
- 调整 SrmPurchaseOrderMapperTest 中的测试用例
- 移除 SrmPurchaseOrderServiceImpl 中未使用的代码

fishSteak committed:
refactor(srm): 重构采购模块相关代码

- 修改字段名：将 approveQty 改为 approvedQty，以统一命名风格
- 优化代码结构：调整部分代码顺序和逻辑，提高可读性
- 移除冗余代码：删除不必要的变量声明和注释
- 修复潜在 bug：修正税价计算逻辑，确保仅在总价和税率均存在时进行计算

fishSteak committed:
refactor(srm): 重构采购入库相关代码

- 修改采购入库单和订单项的版本控制- 优化采购入库单的保存和审核逻辑
- 调整订单项到入库项的转换方式
- 更新相关VO和DO的命名

fishSteak committed:
feat(srm): 优化采购订单项状态管理

- 完善订单项入库和付款后的状态处理逻辑
- 添加订单项自动关闭功能
- 优化错误信息提示
- 调整部分字段名称和描述

fishSteak committed:
refactor(srm): 优化代码格式和结构

- 调整代码缩进和格式，提高可读性
- 移除未使用的变量和导入
-简化部分代码逻辑
- 统一错误码定义格式

fishSteak committed:
refactor(srm): 重构采购入库付款流程

- 新增已付款金额字段和相关方法
- 优化付款状态机逻辑
- 增加订单执行状态机
- 调整订单项状态机
- 重构付款请求和保存请求结构
- 更新相关控制器和服务实现

fishSteak committed:
refactor(srm): 更新权限控制并调整实体类

- 更新了多个控制器中的权限控制注解，将 'erp' 替换为 'srm'
- 调整了 FmsFinanceSubjectDO 实体类的表名和序列注解
- 移除了 SrmOrderInConvert 类中的冗余方法
- 在 SrmPurchaseOrderServiceImpl 中注释掉了不必要的校验逻辑

fishSteak committed:
feat(srm): 优化采购合同模板预热和生成逻辑

- 启用模板预热功能- 优化模板路径解析逻辑- 完善合同生成错误提示
- 修复合同生成过程中资源关闭问题

fishSteak committed:
refactor(srm): 更新采购订单相关接口的权限控制

-将权限控制注解中的权限标识从 'erp:purchasereq-order:*' 修改为 'srm:purchasereq-order:*'- 更新了以下接口的权限控制：
  - 启用/禁用采购订单
  - 合并入库
  - 查询采购合同模板
  - 生成采购合同

fishSteak committed:
refactor(srm): 更新采购订单相关接口的权限标识

-将@srm:purchasereq-order:enable 修改为 @srm:purchase-order:enable
-将@srm:purchasereq-order:merge 修改为 @srm:purchase-order:merge
-将@srm:purchasereq-order:generateContract 修改为 @srm:purchase-order:generateContract

fishSteak committed:
build: 添加部署脚本

fishSteak committed:
test(srm): 禁用采购退货状态机测试

- 在 SrmPurchaseReturnedStatusMachineTest 类上添加 @Disabled 注解，禁用整个测试类

fishSteak committed:
fix(srm): 修复采购模板路径解析问题

- 添加 URLUtil 和 StandardCharsets 引入，用于路径解码
- 新增 getTemplate 方法，用于获取模板
- 在预热模板时，使用 getTemplate 方法统一处理模板获取
- 修复模板路径解析，使用 URLUtil.decode 进行解码，避免中文路径问题

fishSteak committed:
fix 文件放到根包

fishSteak committed:
refactor(srm): 优化采购模板预热机制

- 修改 getTemplate 方法参数名，提高代码可读性
- 使用 Optional 处理文件名，增加代码健壮性
- 过滤非 .docx 文件，避免预热无关文件
- 优化日志输出，使用文件名替代完整路径，提高日志可读性

fishSteak committed:
perf(srm): 优化模板预热机制

- 修改模板预热逻辑，使用异步处理方式- 优化模板加载和 PDF引擎预热的代码结构- 增加日志输出，提高系统可观察性
- 更正配置文件中的模块名称

fishSteak committed:
perf(srm): 优化模板文件预热日志输出

-为 Word 模板预热和 PDF 引擎预热添加耗时记录
- 优化日志输出格式，增加耗时信息

fishSteak committed:
fix(srm): 修改了word的格式

fishSteak committed:
Merge branch 'somle-test' into somle-main-purchase-request

fishSteak committed:
build(srm): 更新状态机相关代码

- 修改状态机相关类和接口的导入路径
- 更新 Action、StateMachine、FailCallback等接口的引用
-调整 SrmPurchaseInItemStatusMachine、SrmPurchaseInStatusMachine 等类中的实现
- 移除 com.alibaba.cola相关依赖，添加 yudao-spring-boot-starter-statemachine 依赖

fishSteak committed:
feat(srm): 添加付款条款功能

- 新增付款条款相关实体类、Mapper、Service、Controller- 实现付款条款的创建、更新、删除、查询等功能
- 添加付款条款的导入导出功能

c-tao commented on yudao-server/src/main/java/cn/iocoder/yudao/server/YudaoServerApplication.java:
```diff
@@ -2,38 +2,39 @@
-@SpringBootApplication(scanBasePackages = {"${yudao.info.base-package}.server", "${yudao.info.base-package}.module","com.somle"})
-@EnableJpaRepositories(basePackages = "com.somle")
-@EntityScan(basePackages = "com.somle")
+@SuppressWarnings("ALL") // 忽略 IDEA 无法识别 ${yudao.info.base-package}
```
启动为什么要改

fishSteak committed:
refactor(srm): 为 SrmPaymentTermSimpleRespVO 类添加 lombok 注解

- 在 SrmPaymentTermSimpleRespVO 类中添加了 @Data 注解
-这个修改可以简化类的代码，自动添加 getter 和 setter 方法

fishSteak committed:
style(infra): 调整代码生成模板的缩进格式- 修改了 controller.vm 文件中的 importExcel 方法
- 优化了方法定义和花括号的缩进方式
- 统一了代码风格，提高了可读性

fishSteak committed:
fix word修改位置

fishSteak committed:
refactor(srm): 优化采购模块状态机处理逻辑

- 新增状态机描述Map，用于记录各种状态机的描述信息
- 修改失败回调逻辑，增加状态机描述信息
- 优化错误信息提示，使其更加详细
- 新增付款条款批量创建功能
- 调整供应商列表查询方式
- 关闭不必要的模板预加载功能

fishSteak committed:
refactor(srm): 优化错误信息并整合 Somle 模块

- 修改采购请求状态错误信息，增加逗号分隔
- 在 YudaoServerApplication 中整合 Somle 模块
-移除 SomleJpaConfig 类，直接在主类中配置 Somle 相关注解

fishSteak committed:
refactor(yudao-server):移除 Somle 配置- 删除了 application-local.yaml 文件中的 somle 配置部分
- 此更改移除了在 local环境中关闭 com.somle 文件夹自动配置的设置

c-tao commented on yudao-dependencies/pom.xml:
```diff
@@ -120,35 +120,46 @@
+            <!--                <scope>test</scope>-->
+            <!--            </dependency>-->
+            <dependency><!--ERP模块状态机-wdy-->
+                <groupId>com.alibaba.cola</groupId>
```
版本号都要使用变量

fishSteak committed:
fix 撤销框架下修改

fishSteak committed:
build(dependencies): 更新依赖版本并添加新依赖

- 添加 cola-component-statemachine、poi-tl、aspose-words、aliyun-sdk-oss 和 commons-codec 依赖
- 更新已存在依赖的版本，使用变量替换硬编码版本号

fishSteak committed:
refactor(srm): 重构模板管理逻辑

- 新增 ErrorCodeConstants 接口，定义通用错误码
- 新增 TemplateManager 工具类，统一模板文件处理逻辑
- 修改 PurchaseOrderTemplateManager，使用新的模板处理方法
- 更新错误码定义，更准确地描述模板处理中的异常情况
-优化模板预热和加载逻辑，提高系统性能和可维护性

fishSteak committed:
refactor(srm): 移除采购订单更新中的异常抛出移除了 SrmPurchaseOrderServiceImpl 类中的 ThrowUtil.ifSqlThrow调用，降低了代码的侵入性。这个改动不影响功能，仅移除了不必要的异常抛出逻辑。

fishSteak committed:
feat(template): 新增模板管理模块

- 新增模板配置属性，支持多个模板路径
- 实现模板预热功能，提高模板加载效率
- 抽离模板策略注册器接口，便于扩展
- 重构模板服务接口，增加模板缓存支持
- 更新应用配置文件，调整模板相关设置

fishSteak committed:
refactor(template): 优化模板预热逻辑

- 将 preloadSingleWordTemplate 方法中的 buildXWPDFTemplate 调用改为 reBuildXWPDFTemplate
- 在 TemplateService 接口中新增 reBuildXWPDFTemplate 方法- 在 TemplateServiceImpl 类中实现 reBuildXWPDFTemplate 方法- 优化 getTemplateBytesFromResource 方法，提取为静态方法

fishSteak committed:
feat(template): 增加模板预热功能并优化模板加载- 新增模板预热配置和功能
- 优化模板加载逻辑，支持 Redis 缓存
- 调整模板扫描路径配置
- 修复部分模板加载问题

fishSteak committed:
test(template): 添加模板服务单元测试

- 新增 TemplateServiceTest 类，实现模板服务的单元测试- 添加测试配置，包括 RedisTemplate 和 ResourcePatternResolver
- 编写测试用例，验证模板字节获取功能
- 优化 TemplateAutoConfiguration 和 TemplateManager
-移除不必要的注释和代码

fishSteak committed:
build:启用工作流、客户管理和 AI 大模型模块- 移除 yudao-module-bpm-biz、yudao-module-crm-biz 和 yudao-module-ai-biz 模块的注释
- 添加这些模块的依赖，以启用相应功能

fishSteak committed:
refactor(template): 优化模板策略预加载逻辑

- 使用 stream 方式替代原有的循环遍历，提高代码可读性和性能
-移除了不必要的条件判断，简化了预加载流程

c-tao committed:
refactor: 使用Resource代替path作为方法入参。重新设计业务模块的模版配置方法。

fishSteak committed:
refactor(template): 重构模板配置和加载机制

- 移除 TemplateRegistar 类，改为使用 TemplateTagPolicyProperty 和 TemplatePolicyRegistrar接口
- 新增 TemplatePurchaseOrderPolicyRegistrarImpl 实现类，处理采购订单模板策略注册
- 更新 TemplateConfigureFactory，使用新的模板策略属性构建配置
- 修改 TemplateService 和 TemplateManager，支持新的模板配置机制
- 更新相关代码，以适应新的模板策略注册和配置方式

fishSteak committed:
refactor(template): 移除 TemplateServiceTest 中未使用的 TemplateProperties 类

移除了 TemplateServiceTest 类中不再使用的 TemplateProperties 导入语句，并调整了 @Import 注解中的导入列表。这个修改有助于清理无用代码，提高代码的可读性和维护性。

c-tao committed:
fix: 命名修改。

c-tao committed:
Merge remote-tracking branch 'origin/somle-main-purchase-request' into somle-main-purchase-request

# Conflicts:
#	yudao-framework/yudao-spring-boot-starter-template/src/test/java/cn/iocoder/yudao/framework/template/core/TemplateServiceTest.java
